examples/basic_authentication/routes/users/_middleware.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:dart_frog/dart_frog.dart';
+import 'package:dart_frog_auth/dart_frog_auth.dart';
+import 'package:example/user_repository.dart';
+
+Future<User?> Function(String, String) userFromCredentials(
+  UserRepository repository,
+) =>
+    (String username, String password) =>
+        repository.userFromCredentials(username, password);
+
+Handler middleware(Handler handler) {
+  final userRepository = UserRepository();
+
+  return handler
+      .use(requestLogger())
+      .use(provider<UserRepository>((_) => userRepository))
+      .use(
+        basicAuthentication<User>(
+          userFromCredentials: userFromCredentials(userRepository),
+          applyToRoute: (RequestContext context) async =>
+              context.request.method != HttpMethod.post,
```

Comment:
I feel like `applyToRoute` is not an ideal name. It is not being applied to a route but to a `Request`. Maybe, we can rename this to simply `applies` or `applyTo`?

Reply:
Related to https://github.com/VeryGoodOpenSource/dart_frog/pull/715#discussion_r1239589315.

Reply:
`applies` indeed feels like a better name, I am gonna change.